FrontEnd/src/components/HeaderFooter/header/navbar/Profile.jsx

```diff
@@ -1,16 +1,57 @@
 import avatar_image from "./Avatar.png";
 import css from "./Profile.module.css";
-import { Link } from 'react-router-dom';
+import Logout from "./Logout";
+import { Link } from "react-router-dom";
+import { useState, useEffect, useRef } from "react";
 
 function Profile(props) {
-    return (
-        <div className={css["header-profile-section"]}>
-            <img className={css["header-profile__avatar"]} src={avatar_image} alt=""/>
-            <div className={css["header-profile-tab"]}>
-                <Link to='/profile/user-info' className={css["header-profile-link__text"]}>Профіль</Link>
-            </div>
-        </div>
-    );
-};
+  const [isOpen, setIsOpen] = useState(false);
+  const ref = useRef(null);
+
+  useEffect(() => {
+    const onBodyClick = (e) => {
+      if (!ref.current.contains(e.target)) {
+        setIsOpen(false);
+      }
+    };
+    document.body.addEventListener("mousedown", onBodyClick);
+
+    return () => {
+      document.body.removeEventListener("mousedown", onBodyClick);
+    };
+  }, []);
```

Comment:
Could we make the dropdown menu as a separate shared component?

Reply:
Done